de.dlr.sc.virsat.graphiti.test/src/de/dlr/sc/virsat/graphiti/diagram/BeanPropertyFactoryTest.java

```diff
@@ -0,0 +1,132 @@
+import de.dlr.sc.virsat.model.dvlm.categories.propertydefinitions.FloatProperty;
+import de.dlr.sc.virsat.model.dvlm.categories.propertydefinitions.IntProperty;
+import de.dlr.sc.virsat.model.dvlm.categories.propertydefinitions.PropertydefinitionsFactory;
+import de.dlr.sc.virsat.model.dvlm.categories.propertydefinitions.ResourceProperty;
+import de.dlr.sc.virsat.model.dvlm.categories.propertydefinitions.StringProperty;
+import de.dlr.sc.virsat.model.dvlm.categories.propertyinstances.APropertyInstance;
+import de.dlr.sc.virsat.model.dvlm.categories.propertyinstances.EReferencePropertyInstance;
+import de.dlr.sc.virsat.model.dvlm.categories.propertyinstances.EnumUnitPropertyInstance;
+import de.dlr.sc.virsat.model.dvlm.categories.propertyinstances.PropertyinstancesFactory;
+import de.dlr.sc.virsat.model.dvlm.categories.propertyinstances.ResourcePropertyInstance;
+import de.dlr.sc.virsat.model.dvlm.categories.propertyinstances.ValuePropertyInstance;
+
+
+public class BeanPropertyFactoryTest {
+
+	@Test
+	public void testBeanPropertyFloatFromFactory() {
+		ValuePropertyInstance vpi = PropertyinstancesFactory.eINSTANCE.createValuePropertyInstance();
+		FloatProperty type = PropertydefinitionsFactory.eINSTANCE.createFloatProperty();
+		vpi.setType(type);
+
+		BeanPropertyFactory bpf = new BeanPropertyFactory();
+		IBeanObject<? extends APropertyInstance> beanPropertyFloat = bpf.getInstanceFor(vpi);
+		assertTrue(beanPropertyFloat instanceof BeanPropertyFloat);
+		assertTrue(beanPropertyFloat.getTypeInstance() instanceof ValuePropertyInstance);
+	}
+
+	@Test
+	public void testBeanPropertyIntFromFactory() {
+		ValuePropertyInstance vpi = PropertyinstancesFactory.eINSTANCE.createValuePropertyInstance();
+		IntProperty type = PropertydefinitionsFactory.eINSTANCE.createIntProperty();
+		vpi.setType(type);
+
+		BeanPropertyFactory bpf = new BeanPropertyFactory();
+		IBeanObject<? extends APropertyInstance> beanPropertyInt = bpf.getInstanceFor(vpi);
+		assertTrue(beanPropertyInt instanceof BeanPropertyInt);
+		assertTrue(beanPropertyInt.getTypeInstance() instanceof ValuePropertyInstance);
+	}
+
+	@Test
+	public void testBeanPropertyBooleanFromFactory() {
+		ValuePropertyInstance vpi = PropertyinstancesFactory.eINSTANCE.createValuePropertyInstance();
+		BooleanProperty type = PropertydefinitionsFactory.eINSTANCE.createBooleanProperty();
+		vpi.setType(type);
+
+		BeanPropertyFactory bpf = new BeanPropertyFactory();
+		IBeanObject<? extends APropertyInstance> beanPropertyBoolean = bpf.getInstanceFor(vpi);
+		assertTrue(beanPropertyBoolean instanceof BeanPropertyBoolean);
+		assertTrue(beanPropertyBoolean.getTypeInstance() instanceof ValuePropertyInstance);
+	}
+
+	@Test
+	public void testBeanPropertyEnumFromFactory() {
+		EnumValueDefinition evd1 = PropertydefinitionsFactory.eINSTANCE.createEnumValueDefinition();
+		EnumProperty enumProperty = PropertydefinitionsFactory.eINSTANCE.createEnumProperty();
+		enumProperty.getValues().add(evd1);
```

Comment:
Do we need to setup enum values for the test case?
If not lets remove this to keep it simple.